Foundation/Process.swift

```diff
@@ -232,23 +232,238 @@ open class Process: NSObject {
+      let listener: SOCKET = socket(AF_INET, SOCK_STREAM, 0)
+      if listener == INVALID_SOCKET {
+        return (first: INVALID_SOCKET, second: INVALID_SOCKET)
+      }
+      defer { closesocket(listener) }
+
+      var result: Int32 = SOCKET_ERROR
+
+      var address: sockaddr_in =
+          sockaddr_in(sin_family: ADDRESS_FAMILY(AF_INET), sin_port: USHORT(0),
+                      sin_addr: IN_ADDR(S_un: in_addr.__Unnamed_union_S_un(S_addr: ULONG("127.0.0.1")!)),
+                      sin_zero: (CHAR(0), CHAR(0), CHAR(0), CHAR(0), CHAR(0), CHAR(0), CHAR(0), CHAR(0)))
+      withUnsafePointer(to: &address) {
+        $0.withMemoryRebound(to: sockaddr.self, capacity: 1) {
+          result = bind(listener, $0, Int32(MemoryLayout<sockaddr_in>.size))
+        }
+      }
+
+      if result == SOCKET_ERROR {
+        return (first: INVALID_SOCKET, second: INVALID_SOCKET)
+      }
+
+      if listen(listener, 1) == SOCKET_ERROR {
+        return (first: INVALID_SOCKET, second: INVALID_SOCKET)
+      }
+
+      withUnsafeMutablePointer(to: &address) {
+        $0.withMemoryRebound(to: sockaddr.self, capacity: 1) {
+          var value: Int32 = Int32(MemoryLayout<sockaddr_in>.size)
+          result = getsockname(listener, $0, &value)
+        }
+      }
+      if result == SOCKET_ERROR {
+        return (first: INVALID_SOCKET, second: INVALID_SOCKET)
+      }
+
+      let first: SOCKET = socket(AF_INET, SOCK_STREAM, 0)
+      if first == INVALID_SOCKET {
+        return (first: INVALID_SOCKET, second: INVALID_SOCKET)
+      }
+
+      var value: u_long = 1
+      if ioctlsocket(first, FIONBIO, &value) == SOCKET_ERROR {
+        closesocket(first)
+        return (first: INVALID_SOCKET, second: INVALID_SOCKET)
+      }
+
+      withUnsafePointer(to: &address) {
+        $0.withMemoryRebound(to: sockaddr.self, capacity: 1) {
+          result = connect(first, $0, Int32(MemoryLayout<sockaddr_in>.size))
+        }
+      }
+
+      if result == SOCKET_ERROR {
+        closesocket(first)
+        return (first: INVALID_SOCKET, second: INVALID_SOCKET)
+      }
+
+      let second: SOCKET = accept(listener, nil, nil)
+      if second == INVALID_SOCKET {
+        closesocket(first)
+        return (first: INVALID_SOCKET, second: INVALID_SOCKET)
+      }
+
+      return (first: first, second: second)
+    }
+    #endif
+
     open func run() throws {
-        
         self.processLaunchedCondition.lock()
         defer {
             self.processLaunchedCondition.broadcast()
             self.processLaunchedCondition.unlock()
         }
 
         // Dispatch the manager thread if it isn't already running
-        
         Process.setup()
-        
+
         // Ensure that the launch path is set
         guard let launchPath = self.executableURL?.path else {
             throw NSError(domain: NSCocoaErrorDomain, code: NSFileNoSuchFileError)
         }
 
+#if os(Windows)
+        // TODO(compnerd) quote the commandline correctly
```

Comment:
apple/swift-llvm has an implementation of this, @jmittert should know where.

Reply:
@jakepetroules - apple/swift-llvm?  Why is that not upstream?  I know that llvm has code for unpacking the parameters, I didn't know that it did quoting too.

Reply:
No worries, it's upstream: https://github.com/llvm-mirror/llvm/blob/6b547686c5410b7528212e898fe30fc7ee7a70a3/lib/Support/Windows/Program.inc#L378

Reply:
@millenomi - does Foundation pull in LLVM to link against?

Reply:
No, we don’t link LLVM other than the language runtimes. You’ll have to port that code over to use it.

Reply:
I think its reasonable to do that as a follow up, what do you think?

Reply:
Please do so, but yup.